### ISSUE TITLE
[BUG] Rspamd milter uses Sendmail's {daemon_name} macro as a value of "MTA-Name" header

### ISSUE THREAD

**Issue by ykasap** [rspamd/rspamd#5017]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Read about bug reporting in general: https://rspamd.com/doc/faq.html#how-to-report-bugs-found-in-rspamd
    * Enabled relevant debugging logs: https://rspamd.com/doc/faq.html#how-to-debug-some-module-in-rspamd 
    * Checked the FAQs about Core files in case of fatal crash: https://rspamd.com/doc/faq.html#how-to-figure-out-why-rspamd-process-crashed
    * Tried ASAN package and obtained the ASAN report (if possible): https://rspamd.com/doc/faq.html#asan-builds 
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+user%3Arspamd
    * Checked that there is not already an experimental package or master branch

**Describe the bug**
When src/libserver/milter.c populates the "MTA-Name" header, it seems to expect that Sendmail doesn't set the {daemon_name} macro, and when {daemon_name} is absent, uses {j} or j macros. Sendmail sets the {daemon_name} macro from DaemonPortOptions Name= suboption, so this logic doesn't work as intended. The "Name=" suboption can be an arbitrary string (defaults to "Daemon#") and used for error messages and logging. 

From Sendmail Installation and Operation Guide (for version 8.18):
>      ${daemon_name}
>           The name of the daemon from DaemonPortOptions
>           Name= suboption.  If this suboption is not set,
>           "Daemon#", where # is the daemon number, is used.

For example, FreeBSD's default configuration is as follows:
```
O DaemonPortOptions=Name=IPv4, Family=inet
O DaemonPortOptions=Name=IPv6, Family=inet6, Modifiers=O
O DaemonPortOptions=Port=587, Name=MSA, M=E
```

I noticed this behavior when I set up the milter_headers module to add the "Authentication-Results" header.
It inserted "IPv4" as authserv-id (which is inappropriate), and I investigated where the string came from. 
```
Authentication-Results: IPv4;	dkim=pass header.d=....
```

Postfix's {daemon_name} is milter_macro_daemon_name (defaults to $myhostname). I believe using Sendmail's {daemon_name} here is inappropriate because it is not expected to be a hostname or something like that.
I don't have a good idea for improving the current code because I don't know a reliable method to distinguish Postfix and Sendmail using defined milter macros. 

A debug output showing these defined milter macros:
```
2024-06-14 13:38:58 #79041(rspamd_proxy) <792307>; milter; rspamd_milter_process_command: got macro command
2024-06-14 13:38:58 #79041(rspamd_proxy) <792307>; milter; rspamd_milter_process_command: got macro: j -> myhostname.example.org
2024-06-14 13:38:58 #79041(rspamd_proxy) <792307>; milter; rspamd_milter_process_command: got macro: _ -> [xx.xx.xx.xx]
2024-06-14 13:38:58 #79041(rspamd_proxy) <792307>; milter; rspamd_milter_process_command: got macro: {daemon_name} -> IPv4
2024-06-14 13:38:58 #79041(rspamd_proxy) <792307>; milter; rspamd_milter_process_command: got macro: {if_name} -> myhostname.example.org
2024-06-14 13:38:58 #79041(rspamd_proxy) <792307>; milter; rspamd_milter_process_command: got macro: {if_addr} -> yy.yy.yy.yy
```

### Steps to Reproduce

1. set up Rspamd as a milter for Sendmail
2. Retrieve a debug output to see {daemon_name} is set.
3. (I don't know how to extract the current MTA-Name value immediately.)

**Expected behavior**
I expect MTA-Name to be a hostname running the sendmail process, calling Rspamd as a milter.

### Versions
% rspamd --version
Rspamd daemon version 3.8.4

CPU architecture x86_64; features: avx, sse2, sse3, ssse3, sse4.1, sse4.2, rdrand
Hyperscan enabled: FALSE
Jemalloc enabled: FALSE
LuaJIT enabled: TRUE (LuaJIT version: LuaJIT 2.1.1710088188)
ASAN enabled: FALSE
BLAS enabled: FALSE
Fasttext enabled: FALSE

FreeBSD 14.1-STABLE stable/14-n267916-3562d64e794b
Sendmail 8.18.1

### Additional Information

Any additional information, configuration or data that might be necessary to reproduce the issue.

**Comment by vstakhov** [comment]
Does sendmail define `{j}` macro by chance?

**Comment by ykasap** [comment]
No, it is not passed by default as follows when using the bundled m4 macros to generate sendmail.cf:
```
O Milter.macros.connect=j, _, {daemon_name}, {if_name}, {if_addr}
O Milter.macros.helo={tls_version}, {cipher}, {cipher_bits}, {cert_subject}, {cert_issuer}
O Milter.macros.envfrom=i, {auth_type}, {auth_authen}, {auth_ssf}, {auth_author}, {mail_mailer}, {mail_host}, {mail_addr}
O Milter.macros.envrcpt={rcpt_mailer}, {rcpt_host}, {rcpt_addr}
O Milter.macros.eom={msg_id}
```

**Comment by ykasap** [comment]
Possibly, macro `j` can be used for both Postfix and Sendmail if `MTA-Name` is supposed to be the MTA's hostname.
One concern is that it might be an incompatible change for some Postfix users because it is always `$myhostname`.
